tensorflow/lite/micro/kernels/ceva/quantize.cc

```diff
@@ -0,0 +1,221 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/lite/kernels/internal/reference/quantize.h"
+
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/quantization_util.h"
+#include "tensorflow/lite/kernels/internal/reference/requantize.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/ceva/ceva_tflm_lib.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/quantize.h"
+#include "tensorflow/lite/micro/micro_utils.h"
+#ifdef MCPS_MEASUREMENT
+#include ""tensorflow/lite/micro/kernels/ceva/mcps_macros.h"
+#endif
+
+namespace tflite {
+namespace {
+
+void* Init(TfLiteContext* context, const char* buffer, size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  return context->AllocatePersistentBuffer(context,
+                                           sizeof(OpDataQuantizeReference));
+}
+
+TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
```

Comment:
This looks like it is identical to the Prepare function in the reference kernel. If that is the case, let's move it to quantize_common.cc similar to SoftmaxPrepare:
https://github.com/tensorflow/tensorflow/blob/b07399e62dd03f6a3b5d8a8adb490601c7c804ee/tensorflow/lite/micro/kernels/softmax_common.cc#L96-L99

I recognize that the one difference here would be that the reference kernel has uint8 support. If you want, you can make a PR that removes uint8 support from quantize or I can do that. I can probably get to this by Friday -- let me know what you would prefer.

Here's one for conv and depthwise_conv: https://github.com/tensorflow/tensorflow/pull/47830

Reply:
Done, see https://github.com/tensorflow/tensorflow/pull/47865